starboard/loader_app/slot_management.cc

```diff
@@ -256,8 +256,10 @@ void* LoadSlotManagedLibrary(const std::string& app_key,
                              LibraryLoader* library_loader,
                              bool use_memory_mapped_file) {
   // Initialize the Installation Manager.
-  SB_CHECK(ImInitialize(kMaxNumInstallations, app_key.c_str()) == IM_SUCCESS)
-      << "Abort. Failed to initialize Installation Manager";
+  if (ImInitialize(kMaxNumInstallations, app_key.c_str()) != IM_SUCCESS) {
```

Comment:
Can we add test case  in the slot_management tests?

Reply:
I'm afraid not. slot_management tests assume the installation manager is initialized, and test the behavior of the installation manager. But at this point, the installation manager is not initialized yet.

Reply:
The setup doesn't do the initialization: 
https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:starboard/loader_app/slot_management_test.cc;l=89?q=slot_man&sq=r:lbshell-internal%2Fcobalt_src%20b:COBALT

You can skip the initialization in the test:
https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:starboard/loader_app/slot_management_test.cc;l=287?q=slot_man&sq=r:lbshell-internal%2Fcobalt_src%20b:COBALT

Reply:
The Setup initializes the slots, but the slots won't be used anyway in this scenario. The scenario is the loader_app tries to initialize the installation manager, but got a NULL returned, so it loads the system image instead. 

I'm afraid I can't think of a way to mock this scenario in the slot_management_test. If you have an idea about how to mock this in the slot_management_test, please suggest.

Reply:
We'll think about other ways to mock the scenario and place the test.